server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/WebAdminServerIntegrationTest.java

```diff
@@ -688,5 +697,34 @@ public void postRedeliverAllEventsShouldCreateATask() {
             .statusCode(HttpStatus.CREATED_201);
     }
 
+    @Test
+    public void postRedeliverGroupEventsShouldCreateATask() {
+
+        String uuid = "6e0dd59d-660e-4d9b-b22f-0354479f47b4";
+        String insertionUuid = "6e0dd59d-660e-4d9b-b22f-0354479f47b7";
+        Group group = new GenericGroup("a");
+        EventDeadLetters.InsertionId insertionId = EventDeadLetters.InsertionId.of(insertionUuid);
+        MailboxListener.MailboxAdded event = EventFactory.mailboxAdded()
+            .eventId(Event.EventId.of(uuid))
+            .user(User.fromUsername(USERNAME))
+            .sessionId(MailboxSession.SessionId.of(452))
+            .mailboxId(InMemoryId.of(453))
+            .mailboxPath(MailboxPath.forUser(USERNAME, "Important-mailbox"))
+            .build();
+
+        guiceJamesServer
+            .getInjector()
+            .getInstance(EventDeadLetters.class)
```

Comment:
why don't you publish it as a probe?

Reply:
what value does it add?

Reply:
"rule of least power"
Giving access to every single classes of james breaks the encapsulation goal of an object oriented design.
As soon as you offer such an API, it will be abused and it will becomes impossible to go forward and make changes to the code because some code will make assumptions about the objects it can have access to and how it behaves.
It's "API design" vs "access everything from everywhere"
In short, publishing the Injector is Evil

Reply:
>  Giving access to every single classes of james breaks the encapsulation goal of an object oriented design.

The access is already granted into any production code through the use of `@Inject` annotation.
It just add access to test code, which can greatly simplify writing some tests. So finally I still don't see the evil part here.

Reply:
Developping probes is a bit verbose but not hard. I don't see why we should refrain nor change the decision for conveniance on that point.

Reply:
I did not say it's hard. And yes verbosity is an inconvenient. I just would like to understand the ins and outs of this decision.

Reply:
In `GuiceJamesServer.java` , `public Injector getInjector() {...}` is public.

This is a major drawback.

Think of the powers we give to people embedding a James guice in another application.

Reply:
see https://github.com/linagora/james-project/pull/2748/commits/0ec7966eecbd49afd55599fbcc510cafaf806b77